### ISSUE TITLE
Distribute source code files in a directory hierarchy

### ISSUE THREAD

**Issue by tekenny** [w7sst/MorseRunner#111]
### Description
Distribute source code files in a directory hierarchy

### Steps To Reproduce
Observe that most files are in the parent directory of the repo

#### Expected behavior
Files distributed in subdirectories to indicate their purpose. Pascal source code, project files, config files, tests, etc.

#### Actual Behavior
NA

#### Reproduces how often
Always

### Version information
  - Morse Runner version: 1.81
  - OS/Version: Windows 10

### Additional context
TASKS
- [ ] determine if it is appropriate to redistribute file in subdirectories and there is consensus to do such (use a poll?)
- [ ] determine which file types should be in what subdirectories
- [ ] fix any problems with the project file to ensure the application can be built
- [ ] updating document in .github subdirectory detailing the purpose of the directory hierarchy

### Can you help?
Please let us know if you are available to help. (replace '[ ]' with '[x]' to affirm)
- [ x ] Yes, I'm available to help test a solution to this problem.

**Comment by w7sst** [comment]
I agree with this change long term. However, we have several branches from other authors what we want to merge (or partially merge) into this project. Once we have completed this merging, then moving code hierarchy will make sense. We want to be careful that we don't make this merging process harder by reorganizing this hierarchy.

Maybe moving all source files down into `src/` would be a first step. Then any other project branches could do the same. The problem with this is that it might be hard to use `git cherry-pick` to apply individual changes. I'm not sure what to recommend here. I do like the idea of improving the source hierarchy in the future.

**Comment by tekenny** [comment]
@w7sst I understand and agree. To track this, should we create a tech debt issue for all the merging needed? Then ref that issue by number as a prerequisite as the first task of this issue. Maybe also create a “deferred” label so that when contributors are looking for new  work they know to skip over deferred. What do you think?

**Comment by w7sst** [comment]
> @w7sst I understand and agree. To track this, should we create a tech debt issue for all the merging needed? Then ref that issue by number as a prerequisite as the first task of this issue. Maybe also create a “deferred” label so that when contributors are looking for new work they know to skip over deferred. What do you think?

The problem is, the various merging will take time to identify and port/merge. We can't wholesale merge these various branches because authors have gone in different directions with lots of differ code reformatting. Some changes we don't want.

I have notes on some of the various branch and will be willing to get them added when I get a change. This is a good idea. I will add a note to do this after I finish the refactoring I'm working on right now.

I agree with adding a `Deferred` label. I notice in our CONTRIBUTING.md file, we have a suggested `Blocked` label that may be equivalent to your suggested `Deferred` label. Our suggested labels are listed [here](https://github.com/w7sst/MorseRunner/blob/main/.github/CONTRIBUTING.md#issue-and-pull-request-labels). Feel free to add these labels to github labels and/or update our label list.

So yes, for now, let's add the `Blocked` (or deferred) label to this issue.

**Comment by tekenny** [comment]
I agree with you. Note that blocked should mean work needs to be performed but can’t often the word impediment is used as well in that case. Deferred just means we know we should/could but not now so it is what you called some day maybe :-)
